solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,400 @@
+          )
+        })
+      )
+    })
+  )
+  default = []
+}
+
+##############################################################################
+
+
+##############################################################################
+# Secondary Interface Variables
+##############################################################################
+
+variable "existing_secondary_subnet_id" {
+  description = "A secondary network interfaces to add to vsi secondary subnets must be in the same zone as VSI. This is only recommended for use with a deployment of 1 VSI."
+  type        = string
+  default     = null
+}
+
+variable "secondary_use_vsi_security_group" {
+  description = "Use the security group created by this module in the secondary interface"
+  type        = bool
+  default     = false
+}
+
+variable "secondary_security_groups" {
+  description = "The security group IDs to add to the VSI deployment secondary interfaces (5 maximum). Use the same value for interface_name as for name in secondary_subnets to avoid applying the default VPC security group on the secondary network interface."
+  type = list(
+    object({
+      security_group_id = string
+      interface_name    = string
+    })
+  )
+  default = []
+}
+
+variable "secondary_floating_ips" {
+  description = "List of secondary interfaces to add floating ips"
+  type        = list(string)
+  default     = []
+}
+
+variable "secondary_allow_ip_spoofing" {
+  description = "Allow IP spoofing on additional network interfaces"
+  type        = bool
+  default     = false
+}
+
+##############################################################################
+
+##############################################################################
+# Snapshot Restore Variables
+##############################################################################
+
+variable "boot_volume_snapshot_id" {
+  description = "The snapshot id of the volume to be used for creating boot volume attachment (if specified, the `image_id` parameter will not be used)"
+  type        = string
+  default     = null
+}
+
+variable "snapshot_consistency_group_id" {
+  description = "The snapshot consistency group Id. If supplied, the group will be queried for snapshots that are matched with both boot volume and attached (attached are matched based on name suffix). You can override specific snapshot Ids by setting the appropriate input variables as well."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+
+variable "use_legacy_network_interface" {
+  description = "Set this to true to use legacy network interface for the created instances."
+  type        = bool
+  nullable    = false
+  default     = false
+}
+
+##############################################################################
+# Dedicated Host Variables
+##############################################################################
+
+variable "enable_dedicated_host" {
```

Comment:
we probably don't need this boolean in the DA. Just use the value of `dedicated_host_id` to determine the boolean value passed to module.